solutions/usaco-664.mdx

```diff
@@ -83,4 +83,44 @@ for i in ans:
 
 </PySection>
 
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.StringTokenizer;
+
+public class BlockGame {
+
+	// Method counts every occurance of characters in a word.
+	// It returns the array containing letter frequency.
+	private static int[] countChars(String word) {
+		int[] freq = new int[26];
+		for (int i = 0; i < word.length(); i++)
+			freq[word.charAt(i) - 'a']++;
+		return freq;
+	}
+
+	// The method selects the maximum value for the frequency of individual letters in words,
+	// then adds the maximum to the sum of letter occurance in answer array.
+	private static void addMaxFreq(int[] answer, int[] freq1, int[] freq2) {
+		for (int i = 0; i < answer.length; i++)
+			answer[i] += Math.max(freq1[i], freq2[i]);
+	}
+
+	// CodeSnip{Kattio}
+
+	public static void main(String[] args) throws IOException {
+		try (Kattio io = new Kattio("blocks")) {
+			int n = io.nextInt();
+			int[] answer = new int[26];
+			for (int i = 0; i < n; i++)
+					addMaxFreq(answer, countChars(io.next()), countChars(io.next()));
+			for (int i : answer) io.println(i);
+		}
```

Comment:
```suggestion
		Kattio io = new Kattio("blocks")
		int n = io.nextInt();
		int[] answer = new int[26];
		for (int i = 0; i < n; i++) {
			addMaxFreq(answer, countChars(io.next()), countChars(io.next()));
		}
		for (int i : answer) {
			io.println(i);
		}
```
no need to `try`; brackets. 
(I edited it on web so u better check again if it really compiles and works

Reply:
I changed, eventually, the try bracketest to be.
I added them earlier because otherwise I would have had to make the io.close() option come to life. From what I've been learning though, developers prefer try with resources. As well as adding brackets, it reduces the risk of errors - you can forget about io.close().
[Link to try with resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html)

Reply:
If the project does not prefer such solution, I will eventually change it.

Reply:
I can only say that this is not a common practice (i.e. I've never seen this in any code form usaco.guide yet). Yes, using `try` might be a good practice in general development, but I'm just not sure if its usage here would improve the readability for the solution. (and certainly it is not conventional for usaco guide)

but @SansPapyrus683 has the last say here.

Reply:
I completely understand, I also see that it reduces readability. I am waiting for the last opinion and I am happy to change

Reply:
don't use try